learn_django/trip_project/views.py

```diff
@@ -1,18 +1,49 @@
+from django.contrib import messages
+from django.shortcuts import render, redirect
+from rest_framework import viewsets, permissions
+from django.contrib.auth.models import Group
 
-from django.shortcuts import render
-from rest_framework import viewsets
-
+from .forms import UserForm, ProfileForm
 from .models import Trip
 from .serializers import TripSerializer
 
 
-class TripView(viewsets.ModelViewSet):
+class IsSuperUserOrStaff(permissions.BasePermission):
+    def has_permission(self, request, view):
+        if request.user and (request.user.groups.filter(name='superuser') or request.user.groups.filter(name='staff')):
+            return True
+        return False
+
+
+class TripsView(viewsets.ModelViewSet):
     queryset = Trip.objects.all()
     serializer_class = TripSerializer
 
+    def get_permissions(self):
```

Comment:
please take a look at `authentication_classes` and `permission_classes` in DRF and see if this code can be refactored or simplified.